pkg/agent/plugin/workloadattestor/docker/docker.go

```diff
@@ -0,0 +1,154 @@
+package docker
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"strings"
+	"sync"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/container"
+	dockerclient "github.com/docker/docker/client"
+	"github.com/hashicorp/hcl"
+	"github.com/spiffe/spire/pkg/agent/common/cgroups"
+	"github.com/spiffe/spire/proto/agent/workloadattestor"
+	"github.com/spiffe/spire/proto/common"
+	spi "github.com/spiffe/spire/proto/common/plugin"
+)
+
+const (
+	_selectorType       = "docker"
+	_subselectorLabel   = "label"
+	_subselectorImageID = "image_id"
+)
+
+// DockerClient is a subset of the docker client functionality, useful for mocking.
+type DockerClient interface {
+	ContainerInspect(ctx context.Context, containerID string) (types.ContainerJSON, error)
+}
+
+type dockerPlugin struct {
+	docker               DockerClient
+	cgroupPrefix         string
+	cgroupContainerIndex int
+	fs                   cgroups.FileSystem
+	mtx                  *sync.RWMutex
+}
+
+type dockerPluginConfig struct {
+	// DockerSocketPath is the location of the docker daemon socket (default: "unix:///var/run/docker.sock" on unix).
+	DockerSocketPath string `hcl:"docker_socket_path"`
+	// DockerVersion is the API version of the docker daemon (default: "1.25").
+	DockerVersion string `hcl:"docker_version"`
+	// CgroupPrefix is the cgroup prefix to look for in the cgroup entries (e.g. "/docker").
+	CgroupPrefix string `hcl:"cgroup_prefix"`
+	// CgroupContainerIndex is the index within the cgroup path where the container ID should be found (default: 0).
+	CgroupContainerIndex int `hcl:"cgroup_container_index"`
+}
+
+func (p *dockerPlugin) Attest(ctx context.Context, req *workloadattestor.AttestRequest) (*workloadattestor.AttestResponse, error) {
+	p.mtx.RLock()
+	defer p.mtx.RUnlock()
+
+	cgroupList, err := cgroups.GetCgroups(req.Pid, p.fs)
+	if err != nil {
+		return nil, err
+	}
+
+	var containerID string
+	var hasDockerEntries bool
+	for _, cgroup := range cgroupList {
+		// We are only interested in cgroup entries that match our desired prefix. Example entry:
+		// "10:perf_event:/docker/2235ebefd9babe0dde4df4e7c49708e24fb31fb851edea55c0ee29a18273cdf4"
+		if !strings.HasPrefix(cgroup.GroupPath, p.cgroupPrefix) {
+			continue
+		}
+		hasDockerEntries = true
+
+		parts := strings.Split(cgroup.GroupPath, "/")
+
+		if len(parts) <= p.cgroupContainerIndex+1 {
+			log.Printf("docker entry found, but without container id: %v", cgroup.GroupPath)
+			continue
+		}
+		containerID = parts[p.cgroupContainerIndex+1]
+		break
+	}
+
+	if !hasDockerEntries {
+		return nil, fmt.Errorf("workloadattestor/docker: no cgroup entries found with prefix %q", p.cgroupPrefix)
```

Comment:
We used to have logging like this in the k8s attestor, but ended up removing it because it would cause the agent to log errors when trying to attest non-k8s workloads. Generally speaking, I think we should support the mix-and-match case well and expect that maybe some dockerized workloads will call us, and some others will call us too... in that vein, perhaps we should not log or return an error here? Curious to hear thoughts on this point

Reply:
Ah okay, I noticed that but wasn't sure of the reason. Agree with this reasoning and updated to reflect